src/Suave.Reloaded/Reloaded.fs

```diff
@@ -0,0 +1,175 @@
+    let ch = ch ()
+    ch
+
+module internal InsideAppDomain =
+
+  module Scripting =
+    open System.Text
+    open System.Threading
+    open System
+    open Suave.Logging
+    open Suave.Http.ServerErrors
+
+    let warn logger = Log.log logger "Suave.Reloaded.Scripting" LogLevel.Warn
+    let info logger = Log.log logger "Suave.Reloaded.Scripting" LogLevel.Info
+
+    type ScriptState =
+      { session : FsiEvaluationSession
+        out     : StringBuilder
+        err     : StringBuilder
+        app     : WebPart
+        logger  : Logger }
+
+      static member Create() =
+        let sbOut = new Text.StringBuilder()
+        let sbErr = new Text.StringBuilder()
+
+        let inStream = new StringReader("")
+        let outStream = new StringWriter(sbOut)
+        let errStream = new StringWriter(sbErr)
+        let fsiConfig = FsiEvaluationSession.GetDefaultConfiguration()
+        let argv = Array.append [|"/fake/fsi.exe"; "--quiet"; "--noninteractive"; "-d:DO_NOT_START_SERVER"|] [||]
+        FsiEvaluationSession.Create(fsiConfig, argv, inStream, outStream, errStream)
+
+    let buildError (config : ScriptConfig) (state : ScriptState) (e:exn) : string =
+      String.Format("""Reloading '{0}' script failed.
+
+Message:
+  {1}
+
+Error:
+  {2}""",
+        config.fsxPath,
+        e.Message,
+        state.err.ToString().Trim())
+
+    let reloadScript (config : ScriptConfig) (state : ScriptState) : ScriptState =
+      try
+        state.session.EvalInteraction(sprintf "#load @\"%s\"" config.fsxPath.FullName)
+        state.session.EvalInteraction(sprintf "open %s" config.modulePath)
+        match state.session.EvalExpression config.expression with
+        | Some app ->
+          { state with app = app.ReflectionValue :?> WebPart }
+
+        | None ->
+          failwith "Couldn't get 'app' value"
+
+      with e ->
+        let state' = ScriptState.Create()
+        { state' with app = INTERNAL_ERROR (buildError config state e) }
+
+    let serverConfig homeFolder =
+      { defaultConfig with
+          homeFolder = Some homeFolder //Some __SOURCE_DIRECTORY__
+          logger     = Loggers.saneDefaultsFor LogLevel.Debug
+          bindings   = [ HttpBinding.mk' HTTP  "127.0.0.1" 8083] }
+
+    let scriptServer (config : ScriptConfig) : Alt<unit> =
+      let rec reconfigure () = job {
+        let home =
+          config.homeFolder |> Option.fold (fun s t ->
+            Path.GetDirectoryName t.fsxPath |> DirectoryInfo)
+
+        let runningState = ScriptState.Create() |> reloadScript (serverConfig homeFolder)
+        let _, server = startWebServerAsync serverConfig state.app
+        let runningServer = Hopac.Extensions.Async.toAlt server |> queue
+        return! loop (runningServer, runningState)
+      }
+      and loop (runningServer, state) : Alt<unit> =
+        let changes = watcher "*.*"
+        Alt.choose [ changes >>% reconfigure
+                     cancellation ]
```

Comment:
The basic loop here seems ok.  However, instead of cancellation or nacks, why not just send a message (in the broad sense) to request termination.  The [`IAsyncDisposable`](http://hopac.github.io/Hopac/Hopac.html#def:type%20Hopac.IAsyncDisposable) documentation might give some ideas.